tests/functional/test_outputter.py

```diff
@@ -0,0 +1,238 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Test level 3 outputter
+"""
+
+from __future__ import (absolute_import, division,
+                        print_function, unicode_literals)
+# pylint:disable=redefined-builtin, unused-wildcard-import, wildcard-import
+from builtins import *
+# pylint:enable=redefined-builtin, unused-wildcard-import, wildcard-import
+from future import standard_library
+standard_library.install_aliases()
+
+from io import StringIO
+
+from aloe.testing import (
+    FeatureTest,
+    in_directory,
+)
+from aloe.result import OutputTerminal
+from mock import patch
+
+
+class MockTermElement(object):
+    """
+    Wraps an element being output by the MockTerminal (i.e. term.bold) and
+    replaces it with a string that can be checked in tests.
+
+    Like with the attributes in Terminal it can be used as a callable or
+    a string.
+    """
+    def __init__(self, attr):
+        self.attr = attr
+
+    def __call__(self, str_):
+        # strips the string to make the output easier to read
+        return 't.{attr}({str})'.format(attr=self.attr,
+                                        str=str_.strip())
+
+    def __str__(self):
+        return '<t.{attr}>'.format(attr=self.attr)
+
+    def __mul__(self, other):
+        return str(self) * other
+
+
+class MockOutputTerminal(OutputTerminal):
+    """Mock terminal to output printable elements instead of ANSI sequences"""
+
+    def __getattr__(self, attr):
+        return MockTermElement(attr)
+
+    def color(self, color):
+        return getattr(self, 'color%s' % color)
+
+
+@in_directory('tests/simple_app')
+class OutputterTest(FeatureTest):
+    """
+    Test level 3 outputter
+    """
+
+    maxDiff = None
+
+    def test_uncolored_output(self):
+        """Test streamed output"""
+
+        with \
+                StringIO() as stream, \
+                patch('aloe.result.AloeTestResult.printSummary'):
+            self.run_features('features/highlighting.feature',
+                              verbosity=3, stream=stream)
+
+            print("--Output--")
+            print(stream.getvalue())
+            print("--END--")
+
+            self.assertEqual(stream.getvalue(), """
+Feature: Highlighting
+
+  As a programmer
+  I want to see my scenarios with pretty highlighting
+  So my output is easy to read
+
+  Scenario: behave_as works                            features/highlighting.feature:12
```

Comment:
I checked again, the front page of Cucumber (https://cucumber.io/) shows lines as comments. Don't bother changing now, can be a separate change later.